File: Module.Frontend.TPM/Client/model/tpm/nonenego/HistoricalNoneNego.js
﻿Ext.define('App.model.tpm.nonenego.HistoricalNoneNego', {
    extend: 'Ext.data.Model',
    idProperty: '_Id',
    breezeEntityType: 'HistoricalNoneNego',

    fields: [
        { name: '_Id', type: 'string', hidden: true },
        { name: '_ObjectId', hidden: true },
        { name: '_User', type: 'string', isDefault: true },
        { name: '_Role', type: 'string', isDefault: true },
        { name: '_EditDate', type: 'date', isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: '_Operation', type: 'string', isDefault: true },
        { name: 'MechanicId', hidden: true, isDefault: true },
        { name: 'MechanicTypeId', hidden: true, isDefault: true, useNull: true, defaultValue: null },
        { name: 'MechanicName', type: 'string', hidden: false, isDefault: true },
        { name: 'MechanicTypeName', type: 'string', hidden: false, isDefault: true },
        { name: 'Discount', type: 'float', hidden: false, isDefault: true },
        { name: 'FromDate', type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'ToDate', type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'CreateDate', type: 'date', hidden: false, isDefault: true, timeZone: +3, convert: dateConvertTimeZone },
        { name: 'ClientTreeFullPathName', type: 'string', hidden: false, isDefault: true },
        { name: 'ProductTreeFullPathName', type: 'string', hidden: false, isDefault: true },
        { name: 'ClientTreeObjectId', type: 'int', hidden: false, isDefault: true },
        { name: 'ProductTreeObjectId', type: 'int', hidden: false, isDefault: true },
        { name: 'ClientTreeId', hidden: true, isDefault: true },
        { name: 'ProductTreeId', hidden: true, isDefault: true }
    ],

    proxy: {
        type: 'breeze',
        resourceName: 'HistoricalNoneNegoes',
        reader: {
            type: 'json',
            totalProperty: 'inlineCount',
            root: 'results'
        }
    }
});

File: Module.Persist.TPM/Model/History/HistoricalNoneNego.cs
using System;
using Module.Persist.TPM.Model.TPM;
using Core.History;

namespace Module.Persist.TPM.Model.History
{
    [AssociatedWith(typeof(NoneNego))]
    public class HistoricalNoneNego : BaseHistoricalEntity<Guid>
    {
        public double? Discount { get; set; }
        public DateTimeOffset? FromDate { get; set; }
        public DateTimeOffset? ToDate { get; set; }
        public DateTimeOffset? CreateDate { get; set; }
        public string MechanicName { get; set; }
        public string MechanicTypeName { get; set; }
        public string ClientTreeFullPathName { get; set; }
        public string ProductTreeFullPathName { get; set; }
        public int ClientTreeObjectId { get; set; }
        public int ProductTreeObjectId { get; set; }
    }
}